### DISCUSSION TITLE
Victron Phoenix 12/3000 230V Smart Wechselrichter

### DISCUSSION THREAD

**Discussion by KarmannColorado** [softwarecrash/Victron2MQTT#24] [General]
Wäre es möglich den Wechselrichter mit einzubinden ? 

Und vielleicht auch das Ladegerät 

Phoenix Smart IP43 Ladegerät 12/50 (3)

Oder gibt es noch nen anderen Weg diese Geräte einzufügen das ich sie sichtbar auf Home Assistant habe ?

Shunt und MPPT laufen bestens ohne Probleme wirklich tolle Arbeit☺️

**Comment by softwarecrash** [comment]
nur wenn die geräte das VE.Direct protokol sprechen und einen anschluss dafür haben.
ansonsten geht es mit dieser software nicht

**Reply by KarmannColorado** [reply]
Ja haben sie, aber sie werden mit der Software von ihnen wohl nicht unterstützt.

**Reply by softwarecrash** [reply]
eigentlich unterstützt die software das komplette ve.direct protokoll.
schwer zu sagen ohne so ein gerät zu haben oder ein log von der datenausgabe

**Reply by KarmannColorado** [reply]
Kann ich als normaler Mensch nen log anfertigen ?

Um es zu zeigen um evtl eine Anpassung zu machen

**Reply by softwarecrash** [reply]
was sagt den die software aktuell wenn du das webserial aufrufst?

**Reply by KarmannColorado** [reply]
MQTT Data Send 

VE Callback Triggered Processing data

CRC Ok 

Das sagt er,mehr aber auch nicht

**Reply by softwarecrash** [reply]
dann bekommt er daten und verarbeitet die auch, über mqtt kommt nichts an? was sagt den http://esp-ip/livejson

**Reply by KarmannColorado** [reply]
Hat geklappt, ich hatte nur gedacht weil keine Daten in der Voransicht standen das es nicht geht 🙈 Funktioniert wunderbar.Danke.. hab auch vor 2 Tagen mal ne Spende da gelassen

**Reply by softwarecrash** [reply]
Vielen Dank,
In der Voransicht stehen nur ein paar ausgewählte Daten, da passt nicht alles hin. Kann sein das dein Gerät nichts von denen liefert, dann ist das natürlich leer, muss ich mir mal Gedanken machen wie man das hübscher lösen kann.

**Reply by KarmannColorado** [reply]
Hab hier nochmal ein Problem wenn ich meinen Solarregler und Inverter anklemme,dann spinnt das System und wechselt immer zwischen Solar und Inverter ..

Jedes Gerät hat einen eigenen ESP D1 Mini

Weiter unten die Logs und ein Video davon. Wäre es machbar den Schalter vom Inverter auch funktionstüchtig zu machen 

Off
On
Eco


Liebe Grüße André 

Video : https://file.io/RrVnwN3fqa30



Victron Inverter

{"ESP_Data":{"IP":"192.168.1.63","sw_version":"1.2.4","ESP_VCC":3.302,"Wifi_RSSI":-59,"Flash_Size":4194304,"Sketch_Size":502688,"Free_Sketch_Space":3670016,"Real_Flash_Size":4194304,"Free_Heap":23056,"HEAP_Fragmentation":10,"WS_Clients":0,"Free_BlockSize":20808,"Runtime":105},"Device_name":"Victron_Inverter","Device_model":"Phoenix Inverter 12V 3000VA 230V","Device_connection":"Connected","Remote_Control_State":false,"Firmware_version_16":"0136","Serial_number":"HQ2126D7TMT","Device_mode":"2","Operation_state":"Inverting","AC_out_volt":230.02,"AC_out_current":0,"AC_out_apparent_pow":"7","Voltage":13.31,"Alarm_code":0,"Warning_reason":"0","Relay":"ON","Off_reason":"None"}





Solar 


{"ESP_Data":{"IP":"192.168.1.93","sw_version":"1.2.4","ESP_VCC":3.353,"Wifi_RSSI":-50,"Flash_Size":4194304,"Sketch_Size":502688,"Free_Sketch_Space":3670016,"Real_Flash_Size":4194304,"Free_Heap":19824,"HEAP_Fragmentation":13,"WS_Clients":2,"Free_BlockSize":17248,"Runtime":1157},"Device_name":"Victron_Solar","Device_model":"SmartSolar MPPT 100|30","Device_connection":"Connected","Remote_Control_State":true,"Firmware_version_16":"164","Serial_number":"HQ22102QGRK","Voltage":13.18,"Battery_current":0,"Panel_voltage":0.01,"Panel_power":0,"Operation_state":"Off","Tracker_operation_mode":"Off","Off_reason":"No input power","Current_error":"No error","Load_output_state":"ON","total_kWh":169.69,"today_kWh":0,"Max_pow_today":0,"Yesterday_kWh":0.07,"Max_pow_yesterday":37,"Day":210}

**Comment by all-solutions** [comment]
Ähm, kann es sein, dass Du beiden das gleiche MQTT-Topic gegeben hast?

**Reply by KarmannColorado** [reply]
Da lag der Fehler ja 🙈 bin noch am Anfang . Danke Danke @softwarecrash  würde man den Schalter vom Inverter noch zum laufen bekommen ? Der hat nämlich keine Funktion über MQTT

**Reply by softwarecrash** [reply]
> Da lag der Fehler ja 🙈 bin noch am Anfang . Danke Danke @softwarecrash  würde man den Schalter vom Inverter noch zum laufen bekommen ? Der hat nämlich keine Funktion über MQTT 

Nein, das ist technisch nicht möglich in dieser Kommunikationsart

**Comment by Cnecne59** [comment]
Zuerst mal ein herzlicher Dank für das tolle Projekt, das doch deutlich mehr Daten anzeigt, als wenn ich mit BLE-Sniffer die Daten auslese!
Ich habe zu dem Inverter noch eine Frage:
In der App gibt es die Möglichkeit den Inverter ein/ und aus zuschalten, ohne das die Verbindung verloren geht (Es wird nur die Last abgeschaltet).
Meiner Meinung nach muss das doch auch über die ve.direkt Verbindung geschaltet werden?
Kann die gleiche Vorgehensweise nicht in das Projekt hier integriert werden?
Denn wenn ich über den eingebauten Anschluss gehe, wird das Gerät komplett ausgeschaltet und auch die BT-Verbindung geht verloren.

Gruß aus Essen
Christoph

**Reply by softwarecrash** [reply]
Das man einfach irgendwas anderes ansteuern geht im ve.direct icht, da es ein oneway system ist.
Du kannst mit der App aber einstellen was passieren soll wenn RX geschaltet wird, schau dazu mal ins wiki

**Reply by Cnecne59** [reply]
Die Einstellung für RX in der App gibt es nur für die MPPT-Regler, in den Einstellungen für den Inverter gibt es die nicht. Ich werde einfach mal versuchen, die Steuerung über den Webhook, bzw. MQTT vorzunehmen.
Ist schon komisch, das es die Methode in der App gibt, den diese ist ja nur über BT mit dem Dongle verbunden, der seinerseits nur mit ve.direct verbunden ist. Das muss also per Software gehen, meine ich.

**Reply by softwarecrash** [reply]
Die bt dongle sind nicht per ve.direct verbunden sondern ve.hex oder ve.can

**Comment by Cnecne59** [comment]
Ach so, dann müsste man das sogar nachbilden, Aufwand sehe ich da! Das hat noch niemand realisiert?
